packages/flutter/lib/src/widgets/sheet.dart

```diff
@@ -0,0 +1,1036 @@
+import 'scroll_position.dart';
+import 'scroll_position_with_single_context.dart';
+import 'scroll_simulation.dart';
+import 'ticker_provider.dart';
+import 'transitions.dart';
+
+// TODO(jamesblasco): Arbitrary values
+const double _kWillPopThreshold = 0.8;
+const Curve _kDecelerateEasing = Cubic(0.0, 0.0, 0.2, 1.0);
+const Duration _kDefaultSheetDuration = Duration(milliseconds: 400);
+const Duration _kBounceAnimationDuration = Duration(milliseconds: 300);
+
+/// NEEDS DOCS
+class SnapSheet extends StatefulWidget {
+
+  const SnapSheet({
+    Key? key,
+    required this.child,
+    this.controller,
+    this.expanded = true,
+    this.draggable = true,
+    this.resized = false,
+    this.bounceAtTop = false,
+  }) : super(key: key);
+
+  /// An object that can be used to control the vertical position of the sheet and
+  /// the position to which the main inner scroll view is scrolled.
+  final SheetController? controller;
+
+  /// The child contained by the [SnapSheet].
+  final Widget child;
+
+  /// Whether the sheet should translate vertically through user interaction
+  ///
+  /// The default value is true.
+  final bool draggable;
+
+  /// Whether the sheet will keep its size when it moves vertically
+  /// or it resizes to fit the remaining space
+  ///
+  /// Used in [CupertinoSheetRoute]
+  ///
+  /// The default value is false.
+  final bool resized;
+
+  /// Whether the widget should bounce when the user drag reaches the top
+  /// limit
+  ///
+  /// Used in [CupertinoSheetRoute]
+  ///
+  /// The default value is false.
+  final bool bounceAtTop;
+
+  /// Whether the widget should expand to fill the available space in its parent
+  /// or not.
+  ///
+  /// In most cases, this should be true. However, in the case of a parent
+  /// widget that will position this one based on its desired size (such as a
+  /// [Center]), this should be set to false.
+  ///
+  /// The default value is true.
+  final bool expanded;
+
+  @override
+  _SheetState createState() => _SheetState();
+
+  /// Called to create the animation controller that will drive the transitions to
+  /// the [SheetRoute] route from the previous one, and back to the previous route from this
+  /// one.
+  static AnimationController createAnimationController(
+    TickerProvider vsync, {
+    Duration? duration,
+  }) {
+    return AnimationController(
+      duration: duration ?? _kDefaultSheetDuration,
+      debugLabel: 'Sheet',
+      vsync: vsync,
+    );
+  }
+}
+
+class _SheetState extends State<SnapSheet> with TickerProviderStateMixin {
+  late SheetController _controller;
+
+  late AnimationController _topBounceController;
+
+  @override
+  void initState() {
+    super.initState();
+    _topBounceController = AnimationController(vsync: this, duration: _kBounceAnimationDuration);
+    _controller = widget.controller ?? SheetController(vsync: this);
+  }
+
+  // TODO(jamesblasco): is there any way to access the Drag inside a GestureDetector and cancel it?
```

Comment:
not using GestureDetector but you can go one level deeper and create the `Drag` object manually and then manage it that way, iirc.